helga_jeopardy.py

```diff
@@ -267,6 +361,170 @@ def scores(client, channel, nick, alltime=False):
+    current_game = mongo_db.find_one({
+        'channel': channel,
+        'game_active': True,
+    })
+
+    for key, value in current_game.items():
+        if key.startswith('cat'):
+            values = [str(value['clue{}'.format(i)]['value']) for i in range(1, 6) if value['clue{}'.format(i)]['active']]
+            cat_names = value["category"]
+            score_response =' '.join([key,  cat_names, ' '.join(sorted(values, key=int))])
+            client.msg(channel, score_response)
+
+    players = current_game['players']
+    client.msg(channel, "Here are the current scores:")
+    for i in players:
+        client.msg(channel, "{} : {}".format(i, current_game[i]))
+
+def evaluate_control(client, channel, nick, current_game, sel_category, sel_value, quest_func=retrieve_question):
+    if current_game["control"] == nick:
+        result = quest_func(client, channel, current_game, sel_category, sel_value, random=False)
+        return result
+    else:
+        client.msg(channel, "You do not have control of the board")
+        return
+
+def check_remaining_clues(client, channel, mongo_db=db.jeopardy):
+    current_game = mongo_db.find_one({
+        'channel': channel,
+        'game_active': True,
+        'game_started': True,
+    })
+    question_count = 0
+    for key, value in current_game.items():
+        if key.startswith('cat'):
+            for k, v in value.items():
+                if k.startswith('clue'):
+                    if v['active'] == True:
+                        question_count += 1
+
+    if question_count < 1:
+        client.msg(channel, "All questions have been answered. Ending game..")
+        end_game(client, channel, current_game)
+        return False
+    else:
+        return True
+
+def end_game(client, channel, current_game=None, mongo_db=db.jeopardy):
+    mongo_db.update({
+	'game_active': True,
+	'channel': channel,
+    }, {
+	'$set': {
+	    'game_active': False,
+	    'game_started': False
+	}
+    })
+    if current_game:
+        current_players = current_game['players']
+        client.msg(channel, "Here are the final scores:")
+        scores = []
+        for i in current_players:
+	    client.msg(channel, "{} : {}".format(i, current_game[i]))
+            scores.append(current_game[i])
+        
+        winners = []
+        for key, value in current_game.items():
+            if value == max(scores):
+                winners.append(key)
+    
+        if len(winners) == 1:
+            client.msg(channel, "{} is our new champion with a score of {}!".format(" ".join(winners), max(scores)))
+            return
+        else:
+            client.msg(channel, "We have a tie! Our winners today are: {}  each of them scored {}".format(" ".join(winners), max(scores)))
+            return
+    else:
+        client.msg(channel, "Game ended by host")
+        return
 
 @command('j', help='usage: ,j [<response>|score]')
 def jeopardy(client, channel, nick, message, cmd, args,
```

Comment:
Consider splitting a lot of this added code out into additional helper functions. Since this is the "entry point" for the entire plugin, keeping this function as small and easy to read/understand as possible would be a huge benefit. In addition, factoring out behavior into smaller helper functions makes that behavior easier to reason about and understand (and even write tests for!).

I would imagine something that delegates to a helper function based on the initial argument(s). Something like:

```
@command(...)
def jeopardy(...):
    # ... reset channel, get question and games, etc. ...
    if len(args) > 0 and args[0] == 'game':
        handle_game_command(...)
    elif len(args) > 0 and args[0] in {'cat1', 'cat2', 'cat3', 'cat4', 'cat5', 'cat6'}:
        handle_category_command(...)
    else:
        handle_answer(...)
```